src/main/java/io/supertokens/inmemorydb/queries/DeleteUserQuery.java

```diff
@@ -0,0 +1,168 @@
+/*
+ *    Copyright (c) 2021, VRAI Labs and/or its affiliates. All rights reserved.
+ *
+ *    This software is licensed under the Apache License, Version 2.0 (the
+ *    "License") as published by the Apache Software Foundation.
+ *
+ *    You may not use this file except in compliance with the License. You may
+ *    obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ *    WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ *    License for the specific language governing permissions and limitations
+ *    under the License.
+ */
+
+package io.supertokens.inmemorydb.queries;
+
+import io.supertokens.inmemorydb.Start;
+import io.supertokens.inmemorydb.config.Config;
+import io.supertokens.pluginInterface.exceptions.StorageQueryException;
+import io.supertokens.pluginInterface.exceptions.StorageTransactionLogicException;
+import io.supertokens.pluginInterface.users.DeleteUserResult;
+
+import javax.annotation.Nonnull;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Collectors;
+
+public class DeleteUserQuery {
+    private final Start start;
+
+    public DeleteUserQuery(Start start) {
+        this.start = start;
+    }
+
+    public DeleteUserResult execute(@Nonnull String userId) throws StorageQueryException, StorageTransactionLogicException {
+        return start.startTransaction(transaction -> {
+            Connection conn = (Connection) transaction.getConnection();
+
+            try {
+                if (!doesUserExist(conn, userId)) {
+                    return DeleteUserResult.failure(DeleteUserResult.FailureReason.UNKNOWN_USER_ID);
+                }
+
+                deleteUserSessionHandles(conn, userId);
+                deleteEmailVerificationData(conn, userId);
+                deleteResetPasswordData(conn, userId);
+                deleteUser(conn, userId);
+                deleteEmailPasswordUser(conn, userId);
+                deleteThirdPartyUser(conn, userId);
+
```

Comment:
Any ideas for how we can make sure that in the future, if we add more tables that need to come in here, we don't forget to add them to this list here?